src/derive.rs

```diff
@@ -285,6 +285,11 @@ pub trait ArgEnum: Sized {
 
     /// Parse an argument into `Self`.
     fn from_str(input: &str, case_insensitive: bool) -> Result<Self, String>;
+
+    /// The canonical argument value.
+    ///
+    /// The value is `None` for skipped variants.
+    fn as_arg(&self) -> Option<&'static str>;
```

Comment:
Isn't it better called as `to_string` or something similar? `strum` does something similar. https://docs.rs/strum/0.21.0/strum/derive.ToString.html

Reply:
- That trait allocates
- I wanted to be clear that this conversion was to the raw argument value and not a general string conversion, which is why `skip`ed values are not covered.  People may choose to implement other traits, like `Display` or `ToString` on top of this though they have to panic or have a sentinel value when rendering a skipped value.